assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -48,148 +44,138 @@ L.Icon.Default.mergeOptions({
   shadowUrl: '/images/marker-shadow.png'
 })
 
-/**
-This Dialog uses the generic Dialog component and combines it with the GeoSearch
-and LocationPopup components as well as a map to display the coordinates on
-It handles setting, displaying, and clearing location information assocaited with a 'street'
- */
-
-class GeotagDialog extends React.Component {
-  static propTypes = {
-    // Provided by react-intl higher-order component
-    intl: PropTypes.object.isRequired,
-
-    // Provided by Redux store
-    street: PropTypes.object,
-    markerLocation: PropTypes.shape({
-      lat: PropTypes.number,
-      lng: PropTypes.number
-    }),
-    addressInformation: PropTypes.object,
-    userLocation: PropTypes.shape({
-      latitude: PropTypes.number,
-      longitude: PropTypes.number
-    }),
-    dpi: PropTypes.number,
+GeotagDialog.propTypes = {
+  // Provided by Redux store
+  street: PropTypes.object,
+  markerLocation: PropTypes.shape({
+    lat: PropTypes.number,
+    lng: PropTypes.number
+  }),
+  addressInformation: PropTypes.object,
+  userLocation: PropTypes.shape({
+    latitude: PropTypes.number,
+    longitude: PropTypes.number
+  })
+}
 
-    // Provided by Redux action dispatchers
-    setMapState: PropTypes.func,
-    addLocation: PropTypes.func,
-    clearLocation: PropTypes.func,
-    saveStreetName: PropTypes.func
+function getInitialState (props) {
+  // Determine initial map center, and what to display
+  let mapCenter, zoom, markerLocation, label
+
+  // If street has a location object, use its position and data
+  if (props.street.location) {
+    mapCenter = props.street.location.latlng
+    zoom = MAP_LOCATION_ZOOM
+    markerLocation = props.street.location.latlng
+    label = props.street.location.label
+    // If we've previously saved marker position, re-use that information
```

Comment:
For this one, I keep forgetting: 

* in what case do we _actually_ expect this condition to be true while the above one is false?

I'd love to make a note of it, or have a test...or strip it out if it isn't anything significant anymore.

Reply:
Ok, gotcha! I thought it was something like that, but I couldn't replicate it. 

Now I see that the reason why is because it only applies if there is no geotagged location. If there is a geotagged location and you repeat the same behavior as above, it _won't_ do this and will go back to the previously selected location. 

My UX sense tells me the behavior should be consistent either way. Either:

* the selected but not confirmed location is always shown if you go about to the Dialog  
or
* if the user exits without confirming, we assume no real choice was made.  

What we have right now is _sometimes_ one behavior happens vs the other.

Reply:
@louh lets follow up on this, maybe once everything else is resolved?